src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1147,6 +1173,145 @@ private static boolean shouldOmitClass(String classname) {
     }
   }
 
+  /**
+   * Returns classes in the given package. Does not include classes in sub-packages. Ignores
+   * non-accessible classes.
+   *
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName}
+   */
+  public static Set<@ClassGetName String> getClassnamesFromPackage(
+      String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+
+    for (File location : getClassLocationsForCurrentClasspath()) {
+      if (location.isFile() && location.getName().endsWith(".jar")) {
+        classnames.addAll(searchInJar(location, packageName, accessibility));
+      } else if (location.isDirectory()) {
+        classnames.addAll(searchInDirectory(location, packageName, accessibility));
+      }
+    }
+
+    return classnames;
+  }
+
+  /**
+   * Searches in the given directory for classes with the given package.
+   *
+   * @param directory a directory
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes with the given package that were found in the specified directory
+   */
+  public static Set<@ClassGetName String> searchInDirectory(
+      File directory, String packageName, AccessibilityPredicate accessibility) {
+    String packageNameAsFile = packageName.replace(".", File.separator) /*.concat(File.separator)*/;
+    File packageDirectory =
+        new File(directory.getPath().concat(File.separator).concat(packageNameAsFile));
+    if (packageDirectory.exists()) {
+      try (Stream<Path> allFiles = Files.walk(Paths.get(packageDirectory.getPath()))) {
+        @SuppressWarnings("signature")
+        Set<@ClassGetName String> classnames =
+            allFiles
+                .filter(e -> e.toFile().isFile()) // to ignore directories
```

Comment:
I find this long sequence of stream operations a bit awkward.  I suspect the code would be cleaner and clearer as a for loop with statements inside.  You might also consider passing a visitor to `walkFileTree`.  However, if you prefer this idiom, that is OK.  Just group the operations so that like functionality is grouped.  For example, currently the filtering occurs in multiple different places.

Reply:
Thanks for your advice! We don`t add classes in subdirectories, so I replaced Files.walk() with for loop called on directory.listFiles() which only shows files for one directory. It allows to skip a lot of extra work, easily filter files and makes code shorter and more readable, so I hope this solution is better than previous.